server/bundles/io.cloudbeaver.service.security/src/io/cloudbeaver/service/security/CBEmbeddedSecurityController.java

```diff
@@ -291,19 +291,76 @@ public SMUser getCurrentUser() throws DBException {
         return getUserById(getUserIdOrThrow());
     }
 
+    @Override
+    public int countUsers(String after, String userIdMask, Boolean enabledState) throws DBCException {
+        try (Connection dbCon = database.openConnection()) {
+            String whereString = "\nWHERE";
+            if (!CommonUtils.isEmpty(after)) {
+                whereString += " USER_ID > ?";
+            }
+            if (!CommonUtils.isEmpty(userIdMask)) {
+                whereString += (whereString != "\nWHERE" ? "AND" : "") + " USER_ID LIKE ?";
+            }
+            if (enabledState != null) {
+                whereString += (whereString != "\nWHERE" ? "AND" : "") + " IS_ACTIVE=?";
+            }
+            try (PreparedStatement dbStat = dbCon.prepareStatement(database.normalizeTableNames(
+                    "SELECT COUNT(*) FROM {table_prefix}CB_USER" + (whereString != "\nWHERE" ? whereString : "")))) {
+                int parameterIndex = 1;
+                if (!CommonUtils.isEmpty(after)) {
+                    dbStat.setString(parameterIndex++, after);
+                }
+                if (!CommonUtils.isEmpty(userIdMask)) {
+                    dbStat.setString(parameterIndex++, "%" + userIdMask + "%");
+                }
+                if (enabledState != null) {
+                    dbStat.setString(parameterIndex++, enabledState ? CHAR_BOOL_TRUE : CHAR_BOOL_FALSE);
+                }
+                try (ResultSet dbResult = dbStat.executeQuery()) {
+                    if (dbResult.next()) {
+                        return dbResult.getInt(1);
+                    } else {
+                        return 0;
+                    }
+                }
+            }
+        } catch (SQLException e) {
+            throw new DBCException("Error while counting users", e);
+        }
+    }
+
     @NotNull
     @Override
-    public SMUser[] findUsers(String userNameMask) throws DBCException {
+    public SMUser[] findUsers(Integer first, String after, String userIdMask, Boolean enabledState)
+            throws DBCException {
         try (Connection dbCon = database.openConnection()) {
             Map<String, SMUser> result = new LinkedHashMap<>();
+            String whereString = "\nWHERE";
+            if (!CommonUtils.isEmpty(after)) {
+                whereString += " USER_ID > ?";
```

Comment:
in my opinion it is strange that we want to compare strings in this way

Reply:
replaced with limit/offset pagination